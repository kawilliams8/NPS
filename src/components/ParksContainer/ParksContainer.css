.ParksContainer {
  background-color: tan;
  display: flex;
  flex-wrap: wrap;
}