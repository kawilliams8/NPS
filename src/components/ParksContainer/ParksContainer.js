import React from 'react';
import Park from '../../containers/Park/Park'

const ParksContainer = ({ parks }) => {

  const parksCards = parks.map(park => {
    return <Park key={park.parkCode} park={park}/>
  })

  return (
    <section className="ParksContainer">
      {parksCards}
    </section>
  )
}

export default ParksContainer;