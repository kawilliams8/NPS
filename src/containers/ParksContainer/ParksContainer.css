.ParksContainer {
  display: flex;
  flex-wrap: wrap;
  background-image: url("../../assets/images/topo.png");
  min-height: 87vh;
}

.parks,
.monts,
.others {
  /* background-color: tan; */
}

.favorites {
  /* background-color: green; */
}

.Park-link {
  background-color: #F5F5F5;
  border-radius: 3px;
  height: 225px;
  margin: 10px;
  object-fit: cover;
  padding: 10px;
  width: 300px;
}