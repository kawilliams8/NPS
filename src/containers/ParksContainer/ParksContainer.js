import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './ParksContainer.css';

export const ParksContainer = (props) => {
  const { type } = props;
  const parkCards = props[type].map(park => {
    const { id, parkCode, name, states, type } = park;
    return (
      <Link to={`/${type}/${parkCode}`}>
        <div className="Park-link" key={id} type={type}>
          <h3>{name}</h3>
          <h4>{states}</h4>
        </div>
      </Link>
    )
  })

  return (
    <section className="ParksContainer">
      {parkCards}
    </section>
  )
}

export const mapStateToProps = state => ({
  parks: state.parks,
  monts: state.monts,
  others: state.others,
  favorites: state.favorites
});

export default connect(mapStateToProps)(ParksContainer);