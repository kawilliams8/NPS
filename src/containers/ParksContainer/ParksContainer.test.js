import React from 'react';
import { shallow } from 'enzyme';
import { ParksContainer, mapStateToProps } from '../ParksContainer/ParksContainer';

describe('ParksContainer', () => {

  let wrapper, mockStore, mockAddFavorite, mockRemoveFavorite;
  beforeEach(() => {
    mockStore = {
      parks: [{
        "states": "DC",
        "directionsInfo": "The memorial is located at the corner of...",
        "directionsUrl": "http://www.nps.gov/afam/planyourvisit/directions.htm",
        "url": "https://www.nps.gov/afam/index.htm",
        "weatherInfo": "Washington DC gets to see all four seasons...",
        "name": "African American Civil War Memorial",
        "latLong": "lat:38.916554, long:-77.025977",
        "description": "Over 200,000 African-American soldiers...",
        "designation": "",
        "parkCode": "afam",
        "id": "1A47416F-DAA3-4137-9F30-14AF86B4E547",
        "fullName": "African American Civil War Memorial",
        "type": "parks"
      },
      {
        "states": "GA",
        "directionsInfo": "To Augusta Canal Discovery Center...",
        "directionsUrl": "http://augustacanal.com/directions.php",
        "url": "https://www.nps.gov/auca/index.htm",
        "weatherInfo": "The weather varies...",
        "name": "Augusta Canal",
        "latLong": "lat:33.5099067687988, long:-82.0053787231445",
        "description": "The Augusta Canal...",
        "designation": "National Heritage Area",
        "parkCode": "auca",
        "id": "813548D1-F8EA-4C07-8878-17FB3C385378",
        "fullName": "Augusta Canal National Heritage Area",
        "type": "parks"
      }],
      monts: [{
        "states": "DC",
        "directionsInfo": "The memorial is located at the corner of...",
        "directionsUrl": "http://www.nps.gov/afam/planyourvisit/directions.htm",
        "url": "https://www.nps.gov/afam/index.htm",
        "weatherInfo": "Washington DC gets to see all four seasons...",
        "name": "African American Civil War Memorial",
        "latLong": "lat:38.916554, long:-77.025977",
        "description": "Over 200,000 African-American soldiers...",
        "designation": "",
        "parkCode": "afam",
        "id": "1A47416F-DAA3-4137-9F30-14AF86B4E547",
        "fullName": "African American Civil War Memorial",
        "type": "parks"
      },
      {
        "states": "GA",
        "directionsInfo": "To Augusta Canal Discovery Center...",
        "directionsUrl": "http://augustacanal.com/directions.php",
        "url": "https://www.nps.gov/auca/index.htm",
        "weatherInfo": "The weather varies...",
        "name": "Augusta Canal",
        "latLong": "lat:33.5099067687988, long:-82.0053787231445",
        "description": "The Augusta Canal...",
        "designation": "National Heritage Area",
        "parkCode": "auca",
        "id": "813548D1-F8EA-4C07-8878-17FB3C385378",
        "fullName": "Augusta Canal National Heritage Area",
        "type": "parks"
      }],
      others: [{
        "states": "DC",
        "directionsInfo": "The memorial is located at the corner of...",
        "directionsUrl": "http://www.nps.gov/afam/planyourvisit/directions.htm",
        "url": "https://www.nps.gov/afam/index.htm",
        "weatherInfo": "Washington DC gets to see all four seasons...",
        "name": "African American Civil War Memorial",
        "latLong": "lat:38.916554, long:-77.025977",
        "description": "Over 200,000 African-American soldiers...",
        "designation": "",
        "parkCode": "afam",
        "id": "1A47416F-DAA3-4137-9F30-14AF86B4E547",
        "fullName": "African American Civil War Memorial",
        "type": "parks"
      },
      {
        "states": "GA",
        "directionsInfo": "To Augusta Canal Discovery Center...",
        "directionsUrl": "http://augustacanal.com/directions.php",
        "url": "https://www.nps.gov/auca/index.htm",
        "weatherInfo": "The weather varies...",
        "name": "Augusta Canal",
        "latLong": "lat:33.5099067687988, long:-82.0053787231445",
        "description": "The Augusta Canal...",
        "designation": "National Heritage Area",
        "parkCode": "auca",
        "id": "813548D1-F8EA-4C07-8878-17FB3C385378",
        "fullName": "Augusta Canal National Heritage Area",
        "type": "parks"
      }],
      favorites: [{
        "states": "DC",
        "directionsInfo": "The memorial is located at the corner of...",
        "directionsUrl": "http://www.nps.gov/afam/planyourvisit/directions.htm",
        "url": "https://www.nps.gov/afam/index.htm",
        "weatherInfo": "Washington DC gets to see all four seasons...",
        "name": "African American Civil War Memorial",
        "latLong": "lat:38.916554, long:-77.025977",
        "description": "Over 200,000 African-American soldiers...",
        "designation": "",
        "parkCode": "afam",
        "id": "1A47416F-DAA3-4137-9F30-14AF86B4E547",
        "fullName": "African American Civil War Memorial",
        "type": "parks"
      },
      {
        "states": "GA",
        "directionsInfo": "To Augusta Canal Discovery Center...",
        "directionsUrl": "http://augustacanal.com/directions.php",
        "url": "https://www.nps.gov/auca/index.htm",
        "weatherInfo": "The weather varies...",
        "name": "Augusta Canal",
        "latLong": "lat:33.5099067687988, long:-82.0053787231445",
        "description": "The Augusta Canal...",
        "designation": "National Heritage Area",
        "parkCode": "auca",
        "id": "813548D1-F8EA-4C07-8878-17FB3C385378",
        "fullName": "Augusta Canal National Heritage Area",
        "type": "parks"
      }]
    };
    mockAddFavorite = jest.fn();
    mockRemoveFavorite = jest.fn();
    wrapper = shallow(<ParksContainer 
      type="parks"
      parks={mockStore.parks}
      addFavorite={mockAddFavorite}
      removeFavorite={mockRemoveFavorite}
      />);
  });
  
  it('should match the parks snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the monts snapshot', () => {
    wrapper = shallow(<ParksContainer
      type="monts"
      monts={mockStore.monts}
      addFavorite={mockAddFavorite}
      removeFavorite={mockRemoveFavorite}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the others snapshot', () => {
    wrapper = shallow(<ParksContainer
      type="others"
      others={mockStore.others}
      addFavorite={mockAddFavorite}
      removeFavorite={mockRemoveFavorite}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the favorites snapshot', () => {
    wrapper = shallow(<ParksContainer
      type="favorites"
      favorites={mockStore.favorites}
      addFavorite={mockAddFavorite}
      removeFavorite={mockRemoveFavorite}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should return an object with parks data from mapStateToProps', () => {
    const mockState = {
      parks: mockStore.parks,
      monts: mockStore.monts,
      others: mockStore.others,
      favorites: mockStore.favorites
    };
    const expected = {
      parks: mockStore.parks,
      monts: mockStore.monts,
      others: mockStore.others,
      favorites: mockStore.favorites
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

});