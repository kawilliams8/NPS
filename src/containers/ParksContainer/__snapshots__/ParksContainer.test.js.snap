// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParksContainer should match the favorites snapshot 1`] = `
<section
  className="ParksContainer favorites"
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/afam"
  >
    <div
      className="Park-link"
      key="1A47416F-DAA3-4137-9F30-14AF86B4E547"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        African American Civil War Memorial
      </h3>
      <h4
        className="Park-link-body"
      />
      <h4
        className="Park-link-body"
      >
        DC
      </h4>
      <p
        className="Park-link-description"
      >
        Over 200,000 African-American soldiers...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/afam"
      >
        Read More...
      </Link>
    </div>
  </Link>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/auca"
  >
    <div
      className="Park-link"
      key="813548D1-F8EA-4C07-8878-17FB3C385378"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        Augusta Canal
      </h3>
      <h4
        className="Park-link-body"
      >
        National Heritage Area
      </h4>
      <h4
        className="Park-link-body"
      >
        GA
      </h4>
      <p
        className="Park-link-description"
      >
        The Augusta Canal...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/auca"
      >
        Read More...
      </Link>
    </div>
  </Link>
</section>
`;

exports[`ParksContainer should match the monts snapshot 1`] = `
<section
  className="ParksContainer monts"
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/afam"
  >
    <div
      className="Park-link"
      key="1A47416F-DAA3-4137-9F30-14AF86B4E547"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        African American Civil War Memorial
      </h3>
      <h4
        className="Park-link-body"
      />
      <h4
        className="Park-link-body"
      >
        DC
      </h4>
      <p
        className="Park-link-description"
      >
        Over 200,000 African-American soldiers...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/afam"
      >
        Read More...
      </Link>
    </div>
  </Link>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/auca"
  >
    <div
      className="Park-link"
      key="813548D1-F8EA-4C07-8878-17FB3C385378"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        Augusta Canal
      </h3>
      <h4
        className="Park-link-body"
      >
        National Heritage Area
      </h4>
      <h4
        className="Park-link-body"
      >
        GA
      </h4>
      <p
        className="Park-link-description"
      >
        The Augusta Canal...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/auca"
      >
        Read More...
      </Link>
    </div>
  </Link>
</section>
`;

exports[`ParksContainer should match the others snapshot 1`] = `
<section
  className="ParksContainer others"
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/afam"
  >
    <div
      className="Park-link"
      key="1A47416F-DAA3-4137-9F30-14AF86B4E547"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        African American Civil War Memorial
      </h3>
      <h4
        className="Park-link-body"
      />
      <h4
        className="Park-link-body"
      >
        DC
      </h4>
      <p
        className="Park-link-description"
      >
        Over 200,000 African-American soldiers...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/afam"
      >
        Read More...
      </Link>
    </div>
  </Link>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/auca"
  >
    <div
      className="Park-link"
      key="813548D1-F8EA-4C07-8878-17FB3C385378"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        Augusta Canal
      </h3>
      <h4
        className="Park-link-body"
      >
        National Heritage Area
      </h4>
      <h4
        className="Park-link-body"
      >
        GA
      </h4>
      <p
        className="Park-link-description"
      >
        The Augusta Canal...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/auca"
      >
        Read More...
      </Link>
    </div>
  </Link>
</section>
`;

exports[`ParksContainer should match the parks snapshot 1`] = `
<section
  className="ParksContainer parks"
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/afam"
  >
    <div
      className="Park-link"
      key="1A47416F-DAA3-4137-9F30-14AF86B4E547"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        African American Civil War Memorial
      </h3>
      <h4
        className="Park-link-body"
      />
      <h4
        className="Park-link-body"
      >
        DC
      </h4>
      <p
        className="Park-link-description"
      >
        Over 200,000 African-American soldiers...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/afam"
      >
        Read More...
      </Link>
    </div>
  </Link>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/parks/auca"
  >
    <div
      className="Park-link"
      key="813548D1-F8EA-4C07-8878-17FB3C385378"
      type="parks"
    >
      <h3
        className="Park-link-heading"
      >
        Augusta Canal
      </h3>
      <h4
        className="Park-link-body"
      >
        National Heritage Area
      </h4>
      <h4
        className="Park-link-body"
      >
        GA
      </h4>
      <p
        className="Park-link-description"
      >
        The Augusta Canal...
      </p>
      <p
        className="Park-link-ellipsis"
      >
        ...
      </p>
      <Link
        className="Park-link-more"
        to="/parks/auca"
      >
        Read More...
      </Link>
    </div>
  </Link>
</section>
`;
